plugins/csv/src/main/java/org/itsallcode/whiterabbit/plugin/csv/CSVProjectReportExporter.java

```diff
@@ -0,0 +1,100 @@
+package org.itsallcode.whiterabbit.plugin.csv;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.text.MessageFormat;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.itsallcode.whiterabbit.api.features.ProgressMonitor;
+import org.itsallcode.whiterabbit.api.features.ProjectReportExporter;
+import org.itsallcode.whiterabbit.api.model.DayType;
+import org.itsallcode.whiterabbit.api.model.ProjectReport;
+import org.itsallcode.whiterabbit.api.model.ProjectReportActivity;
+import org.itsallcode.whiterabbit.api.model.ProjectReportDay;
+
+class CSVProjectReportExporter implements ProjectReportExporter
+{
+    private final boolean filterForWeekDays;
+    private final OutStreamProvider outStreamProvider;
+    private final String separator;
+
+    CSVProjectReportExporter(boolean filterForWeekDays, String separator, OutStreamProvider outStreamProvider) {
+        this.filterForWeekDays = filterForWeekDays;
+        this.outStreamProvider = outStreamProvider;
+        this.separator = separator;
+    }
+
+    @Override
+    public void export(ProjectReport report, ProgressMonitor progressMonitor)
+    {
+        progressMonitor.setTaskName("Exporting...");
+        if (progressMonitor.isCanceled())
+        {
+            return;
+        }
+        List<ProjectReportDay> filteredDays = filterForWeekDays ?
+                report.getDays().stream().filter(day -> day.getType() == DayType.WORK).collect(Collectors.toList()) :
+                report.getDays();
+
+        progressMonitor.beginTask("Initializing...", report.getDays().size());
```

Comment:
Updating the progress monitor is not necessary here, because the export will be very fast.
This was introduced for the pm-smart export which takes some time.